mongo/src/main/java/com/redhat/lightblue/mongo/crud/MongoCRUDController.java

```diff
@@ -785,8 +784,7 @@ private void createUpdateEntityInfoIndexes(EntityInfo ei, Metadata md) {
                 if (index.getName() != null && index.getName().trim().length() > 0) {
                     options.append("name", index.getName().trim());
                 }
-                // if we have hidden fields to generate, we want index creation to be blocking so we can ensure that the indexes are created before we generate the fields
-                options.append("background", !hidden);
+                options.append("background", true);
```

Comment:
I don't remember, but we probably thought inserting a doc while building an
index wouldn't work. Why would we have thought that? Maybe things happened?

On Tue, May 24, 2016 at 12:39 PM, Brandon Vulaj <notifications@github.com>
wrote:

> In
> mongo/src/main/java/com/redhat/lightblue/mongo/crud/MongoCRUDController.java
> <https://github.com/lightblue-platform/lightblue-mongo/pull/283#discussion_r64446507>
> :
>
> > @@ -785,8 +784,7 @@ private void createUpdateEntityInfoIndexes(EntityInfo ei, Metadata md) {
> >                  if (index.getName() != null && index.getName().trim().length() > 0) {
> >                      options.append("name", index.getName().trim());
> >                  }
> > -                // if we have hidden fields to generate, we want index creation to be blocking so we can ensure that the indexes are created before we generate the fields
> > -                options.append("background", !hidden);
> > +                options.append("background", true);
>
> I'm trying to remember - why did we have this blocking in the first place?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/lightblue-platform/lightblue-mongo/pull/283/files/1461bf80e101a3ba0186e14370650a1408981ab7#r64446507>
>